src/main/java/adris/altoclef/tasks/resources/wood/CollectSticksTask.java

```diff
@@ -0,0 +1,139 @@
+            for (Item plank : ItemHelper.PLANKS) {
+                int count = mod.getItemStorage().getItemCount(plank);
+                if (count > 0) {
+                    setDebugState("Crafting sticks from planks");
+                    return new CraftInInventoryTask(new RecipeTarget(Items.STICK, (count/2)*4, generateStickRecipe(ItemHelper.PLANKS)));
+                }
+            }
+        }
+        // Do the same for logs to craft planks
+        int potentialStickCountFromLogsAndPlanks = potentialStickCountFromPlanks + mod.getItemStorage().getItemCount(ItemHelper.LOG)*8;
+        if (potentialStickCountFromLogsAndPlanks >= _targetCount) {
+            for (Item log : ItemHelper.LOG) {
+                int count = mod.getItemStorage().getItemCount(log);
+                if (count > 0) {
+                    Item plankCheck = ItemHelper.logToPlanks(log);
+                    if (plankCheck == null) {
+                        Debug.logError("Invalid/Un-convertable log: " + log + " (failed to find corresponding plank)");
+                    }
+                    int plankCount = mod.getItemStorage().getItemCount(plankCheck);
+                    int otherPlankCount = mod.getItemStorage().getItemCount(ItemHelper.PLANKS) - plankCount;
+                    int targetTotalPlanks = Math.min(count*4 + plankCount, _targetCount - otherPlankCount);
+                    setDebugState("Crafting planks from logs for sticks");
+                    return new CraftInInventoryTask(new RecipeTarget(plankCheck, targetTotalPlanks, generatePlankRecipe(ItemHelper.LOG)));
+                }
+            }
+        }
+        // Check the ground for loot
+        if (mod.getEntityTracker().itemDropped(_itemsToPickup)) {
+            return new PickupDroppedItemTask(new ItemTarget(_itemsToPickup), true);
+        }
+        // Mine wood or dead bushes
+        ArrayList<Block> blocksToMine = new ArrayList<>(Arrays.asList(ItemHelper.itemsToBlocks(ItemHelper.LOG)));
+        blocksToMine.add(Blocks.DEAD_BUSH);
+        blocksToMine.addAll(Arrays.asList(ItemHelper.itemsToBlocks(ItemHelper.PLANKS)));
+        Optional<BlockPos> block = mod.getBlockTracker().getNearestTracking(ItemHelper.itemsToBlocks(_itemsToPickup)); // very janky but i suck at java
+        if (block.isPresent()) {
+            BlockPos blockPos = block.get();
+            return new DestroyBlockTask(blockPos);
+        }
+        return new TimeoutWanderTask();
+    }
+
+    @Override
+    protected void onResourceStop(AltoClef mod, Task interruptTask) {
+        mod.getBlockTracker().stopTracking(Blocks.DEAD_BUSH);
+        mod.getBlockTracker().stopTracking(ItemHelper.itemsToBlocks(ItemHelper.LOG));
+        mod.getBlockTracker().stopTracking(ItemHelper.itemsToBlocks(ItemHelper.PLANKS));
+        mod.getBehaviour().pop();
+    }
+
+    @Override
+    protected boolean isEqualResource(ResourceTask other) {
+        return other instanceof CollectSticksTask;
```

Comment:
Also make sure the counts are the same.

Reply:
we don't need to check for the count if the class is refactored. See CollectHaybailTask for more info.